### DISCUSSION TITLE
`The head ref may contain hidden characters` falsely appears on CJK characters

### DISCUSSION THREAD

**Discussion by scarf005** [community/community#51906] [Pull Requests]
### Select Topic Area

Question

### Body

![image](https://user-images.githubusercontent.com/54838975/229679524-7d2eb039-cbcb-4fdc-819d-2e1d33b8159f.png)

iirc hidden charcters are something like Nbsp, so looks like this feature is causing false positive on CJK characters.

**Comment by enslo** [comment]
I'm Japanese, and get the same warning.

I haven't done anything to be warned about it. One day it suddenly appeared.

Although there's a slight chance that my repository is contaminated with hidden characters, but at least, it doesn't contain any nbsp.

I suspect the warning comes from CJK characters for some reason. Perhaps a Japanese branch name?

**Comment by gkitzing** [comment]
It is not only CJK characters. Happens with my pull-requests too, when using Umlauts (äöüÄÖÜß).

Strange message, although nothing seems to stop the PR from being merged.

**Comment by vvtommy** [comment]
I received the same warning when using Chinese characters.

**Comment by EricDunsworth** [comment]
I also get the warning with other kinds of accented Latin characters (like ``é`` in French).

**Comment by tarper24** [comment]
I'll chime in and say that I got this same warning for emoji, so it sounds like it's triggered for any non-ascii Unicode characters

**Comment by ryush00** [comment]
+1

**Comment by tom-id-au** [comment]
Yep, also just happens for a long dash (—) that was auto-created by two normal dashes (--) by my git client (Tower on macOS)

**Comment by SV-Chen-Jun** [comment]
Any suggestions to fix this issue?

**Comment by josephfrazier** [comment]
I got the same for a branch name with double quote (`"`) characters in it

**Comment by alexandre-machado** [comment]
in Portuguese character too
![image](https://github.com/community/community/assets/1766903/fb80d106-3ef6-4da0-b875-3b21896d2e92)

**Comment by Repiteo** [comment]
If this is happening because of the potential for glyphs a user's fonts can't display, wouldn't that be solved by displaying a fallback character instead? That's certainly preferable over making any non-ASCII character throw an extremely misleading warning

**Comment by cheeselemon** [comment]
Wow still unfixed?

**Comment by golee** [comment]
Calling perfectly normal text “hidden characters” is inaccurate and culturally offensive.
